src/actions/GameActions.tsx

```diff
@@ -1,20 +1,39 @@
+import { EffectName, StatusEffect } from 'src/GamePlay/Effect';
+import { StoreState } from 'src/types/StoreState';
+import { TypedAction } from "redoodle";
 import { ThunkAction } from 'redux-thunk';
-import { ofType, unionize, UnionOf } from 'unionize';
-import { EffectName } from 'src/GamePlay/Effect';
-import { StoreState, Enemy } from 'src/types/StoreState';
-
-export const Actions = unionize({
-    ADJUST_HP: ofType<{ targetEntityId: string, hp: number }>(),
-    ADJUST_ENERGY: ofType<{ energy: number }>(),
-    SET_ENERGY: ofType<{ energy: number }>(),
-    APPLY_EFFECT: ofType<{ effectName: EffectName, targetId: string, magnitude: number }>(),
-    CLEAR_ENEMIES: ofType<{}>(),
-    ADD_ENEMY: ofType<Partial<Enemy>>(),
-    REMOVE_ENEMY: ofType<string>(),
-}, {
-    tag: 'type',
-    value: 'payload',
-});
-
-export type ActionsType = UnionOf<typeof Actions>;
-export type ThunkType = ThunkAction<void, StoreState, void, ActionsType>
\ No newline at end of file
+
+export const AdjustHp = TypedAction.define("gameplay::adjustHp")<{ 
```

Comment:
I do kind of prefer SCREAMING_SNAKE_CASE, as it's very clear when an action is happening vs just another function invocation.